airflow/metrics/datadog_logger.py

```diff
@@ -51,7 +55,33 @@
     )
 
 
-class SafeDogStatsdLogger:
+def prepare_stat_with_tags(fn: T) -> T:
+    """Prepare tags and stat."""
+
+    @wraps(fn)
+    def wrapper(self, stat: str | None = None, *args, tags: dict[str, str] | None = None, **kwargs):
+        stat = stat or ""
```

Comment:
Why do we need this `stat or ""`? Shouldn't this decorator pass through whatever it's given?

Reply:
Because in the StatsLogger all of the functions get "str" as metric_name and not "str | None"

Reply:
Not _all_ of them though -- timer/timing accepts None doesn't it?

Reply:
Yeah we decided to let only timer/timing accept None. Is it something that need to change

Reply:
Also in the new commit I deleted them both from datalog and statd.